### ISSUE TITLE
Usage is broken in Avalonia 11

### ISSUE THREAD

**Issue by alex6dj** [mysteryx93/HanumanInstitute.MvvmDialogs#4]
Seems that Avalonia v11 have some breaking changes in the use of [IDispatcher](https://github.com/AvaloniaUI/Avalonia/issues/9024). Any workaround is welcome.

```
Method not found: 'System.Threading.Tasks.Task`1<System.__Canon> Avalonia.Threading.IDispatcher.InvokeAsync(System.Func`1<System.__Canon>, Avalonia.Threading.DispatcherPriority)'.
```

```
   at HanumanInstitute.MvvmDialogs.Avalonia.DialogManager.DispatchAsync[T](Func`1 action)
   at HanumanInstitute.MvvmDialogs.DialogManagerBase`1.ShowFrameworkDialogAsync[TSettings](INotifyPropertyChanged ownerViewModel, TSettings settings, AppDialogSettingsBase appSettings, Func`2 resultToString)
   at HanumanInstitute.MvvmDialogs.DialogServiceExtensions.ShowOpenFolderDialogAsync(IDialogService service, INotifyPropertyChanged ownerViewModel, OpenFolderDialogSettings settings, AppDialogSettingsBase appSettings)
   at MultiConverter.ViewModels.Options.OptionsViewModel.UpdateTemporalFolderPath(IDialogService dialogService) in F:\!Cosas\source\repos\MultiConverter\src\MultiConverter\ViewModels\Options\OptionsViewModel.cs:line 97
```

Windows 10
Avalonia 11 preview 2

**Comment by mysteryx93** [comment]
I'm working on a new version for v11.

There's a whole new file picker API, and adding support for mobile navigation.

The current Git version should already work for existing features.

**Comment by alex6dj** [comment]
Good to know. Right now my project is in his firsts steps, so using MvvmDialogs for folder selection only so no Dialogs yet. Im not sure what do you mean with "Git version", I think you refer to build from the source, but Im more a nuget update approach guy :wink:.

A fast question, in windows what kind of folder dialog is opened, I know there are at least two. FolderBrowserDialog, I dont like this guy, is simple but not good when working with network folders (my case). And the other (Dont know the name), more similar to the open file dialog, more complex but better in all ways. Maybe there is a way to use one over the other.

**Comment by mysteryx93** [comment]
That's handled by Avalonia itself, and varies with the OS. If you got an issue with the dialog opened by Avalonia on Windows, you can open a ticket on Avalonia repo.

You're better to use stable Avalonia 0.10.11 until the new package is released.

As a difference between the current version and new version, the browser dialog will now return an object instead of a string.

You can simply replace 

    var path = await dialogService.ShowOpenFolderDialogAsync(...);

with

    var result = await dialogService.ShowOpenFolderDialogAsync(...);
    var path = result.Path;

Unless you want cross-platform support on platforms that run the app in a container, then you'll need to use the new API on the return object.

On mobile phones, you'll get a bookmark to the file as a string that you can save and re-use later; as you otherwise can't access the file directly.

**Comment by alex6dj** [comment]
OK, thanks for the answer and for your time. You can close this issue, or better let it open until full Avalonia v11 support is finished.

**Comment by FSharpCSharp** [comment]
@mysteryx93 
What is the status for version 11? There is already a 2.0 preview version in the repository. Unfortunately, this is not (yet) available via NuGet, and the last two months have been very quiet. Therefore I wanted to ask here if an official preview version is to be expected and if the project is still being actively developed here. Would be really desirable, because I think it would be great if a universal solution to the problem dialogs for Avalonia is available. You don't have to reinvent the wheel over and over again.

**Comment by mysteryx93** [comment]
The GitHub code should work for desktop (mobile not yet tested/completed).

Then I got really swamped with other things and couldn't get to finish this. As soon as I get the chance, I'll get back to complete this. Can't say when.

**Comment by mysteryx93** [comment]
have you tried it? Has anyone seen any issues on desktop with the Git version?

If not, I could just publish the current code as-in as preview

It has very considerable internal structural changes, so it's possible that it has unresolved issues.

**Comment by FSharpCSharp** [comment]
> have you tried it? Has anyone seen any issues on desktop with the Git version?
> 
> If not, I could just publish the current code as-in as preview
> 
> It has very considerable internal structural changes, so it's possible that it has unresolved issues.

I have firmly resolved to test the coming days. But unfortunately I have not yet come to it, I'm sorry. But it would be really great if you could provide a general preview version, which could then be accessed via NuGet. Of course I could do that for a short test directly from the repo, but it's easier that way.

**Comment by mysteryx93** [comment]
I've updated dependencies and released package v2.0-preview1 with the code as-is.

Let me know if you run into issues.

The back-end code structure had to go through refactoring to support a new navigation mode for mobile devices, although that work hasn't been completed. If you run into issues with the updated code, let me know.
